### ISSUE TITLE
Analyics logEvent KTX should support Parcelable and Parcelable[]

### ISSUE THREAD

**Issue by samtstern** [firebase/firebase-android-sdk#1420]
## What feature would you like to see?

Right now the `.logEvent(name, ParameterBuilder)` ktx only supports primitive types.

## How would you use it?

I would like to be able to do this:
```kotlin
        val removeCartParams = Bundle().apply {
            putString(FirebaseAnalytics.Param.CURRENCY, "USD")
            putDouble(FirebaseAnalytics.Param.VALUE, 1 * 24.99)
            putParcelableArray(FirebaseAnalytics.Param.ITEMS, arrayOf<Parcelable>(itemBootsCart))
        }

        analytics.logEvent(FirebaseAnalytics.Event.REMOVE_FROM_CART, removeCartParams)
```

As this
```kotlin
        analytics.logEvent(FirebaseAnalytics.Event.REMOVE_FROM_CART) {
            param(FirebaseAnalytics.Param.CURRENCY, "USD")
            param(FirebaseAnalytics.Param.VALUE, 1 * 24.99)
            param(FirebaseAnalytics.Param.ITEMS, arrayOf<Parcelable>(itemBootsCart))
        }
```

**Comment by samtstern** [comment]
@vkryachko I would have sent a PR but doesn't look like the Analytics KTX source is here?  IMO this should not require a new API review, it's more like an omission than a change.

**Comment by vkryachko** [comment]
We might have left it out intentionally, @g-charbel to confirm.

Additionally isn't `Parcelable` too permissive here? Should we restrict it to `Bundle` instead?

**Comment by samtstern** [comment]
I agree it's too permissive but the SDK docs use `Parcelable`:
https://firebase.google.com/docs/reference/android/com/google/firebase/analytics/FirebaseAnalytics.Param#ITEMS

**Comment by vkryachko** [comment]
I agree that it says that, but I strongly suspect that this is because `Bundle`s don't have the `putBundleArray()` method, however afaict if I put anything other than a bundle I will get an error or a silently dropped param/event.

Since we don't have that limitation here I suggest we add an overload like:

```kotlin
  fun param(name: String, vararg values: Bundle)
```

That will not force me to create an array or list explicitly and if I have one I can always call it via "spread": `param("foo", *myBundleList)`. wdyt?

**Comment by samtstern** [comment]
Ok that sounds good to me I think.  So what would it look like for my example above?

**Comment by vkryachko** [comment]
OK, trying to represent your example above uncovered an issue with the vararg approach. The fact that your list contains just one element would mean that the wrong overload would be picked:

```kotlin
param(name: String, value:Bundle) // in your case this one will get picked
param(name: String, vararg values: Bundle)
```

So the only way to resolve this ambiguity would be to explicitly call `param("items", *arrayOf(item))` which might be confusing.

So I think it would be better to use your original idea but restrict to Bundle:

```kotlin
param(name: String, value: Array<Bundle>)
```

**Comment by vkryachko** [comment]
@G-charbel we already have primitive types, in kotlin they are represented as `Int`, `Boolean`, etc, while `Int?` are boxed nullable ones.

**Comment by vkryachko** [comment]
This is code-complete and will release with the next release of analytics.

**Comment by G-charbel** [comment]
Got it!

**Comment by vkryachko** [comment]
Released in 17.4.0
